src/client-c++/cli_main.cpp

```diff
@@ -0,0 +1,410 @@
+        }
+    }
+    return sNumRequiredArgsFound == NumRequiredArgs;
+}
+
+void PrintHelp(const std::string& programNameParm)
+{
+    std::size_t sLongestOption = 0;
+    for(std::size_t sIdx = 0; sIdx < NumOptOptions; sIdx++)
+    {
+        std::size_t sLength = strlen(create_long_options[sIdx].name);
+        sLongestOption      = std::max(sLength, sLongestOption);
+    }
+
+    std::cout << "Usage: " << programNameParm << std::endl;
+    std::cout << "Required:" << std::endl;
+    for(std::size_t sIdx = 0; sIdx < NumOptOptions; sIdx++)
+    {
+        // Assumes that the required args always are grouped first in the list
+        if(NumRequiredArgs == sIdx)
+        {
+            std::cout << std::endl << "Optional:" << std::endl;
+        }
+        std::size_t sCurrentOptionLength = strlen(create_long_options[sIdx].name);
+        std::cout << "\t-" << (char)create_long_options[sIdx].val << " -"
+                  << create_long_options[sIdx].name;
+        for(std::size_t sPadIdx = 0; sPadIdx < (sLongestOption - sCurrentOptionLength); sPadIdx++)
+        {
+            std::cout << " ";
+        }
+        std::cout << " --" << create_long_options[sIdx].name;
+        for(std::size_t sPadIdx = 0; sPadIdx < (sLongestOption - sCurrentOptionLength); sPadIdx++)
+        {
+            std::cout << " ";
+        }
+        std::cout << " | " << OptOptionsHelpText[sIdx] << std::endl;
+    }
+}
+
+void Verbose_PrintArgs(const SfClientArgs& argsParm)
+{
+    std::cout << "Project:            " << argsParm.mProject << std::endl;
+    std::cout << "Comment:            " << argsParm.mComment << std::endl;
+    std::cout << "Epwd Path:          " << argsParm.mEpwdPath << std::endl;
+    std::cout << "URL:                " << argsParm.mUrl << std::endl;
+    std::cout << "Private Key Path:   " << argsParm.mPrivateKeyPath << std::endl;
+    std::cout << "Payload Path:       " << argsParm.mPayloadPath << std::endl;
+    std::cout << "Extra Server Parms: " << argsParm.mExtraServerParms << std::endl;
+    std::cout << "Output Path:        " << argsParm.mOutputPath << std::endl;
+    std::cout << "Password ENV:        " << argsParm.mPasswordEnvVar << std::endl;
+    std::cout << std::endl;
+    std::cout << "StdOut:   " << (argsParm.mServerStdOut ? "true" : "false") << std::endl;
+    std::cout << "Verbose:  " << (argsParm.mVerbose ? "true" : "false") << std::endl;
+    std::cout << "Debug:    " << (argsParm.mDebug ? "true" : "false") << std::endl;
+}
+
+int main(int argc, char** argv)
+{
+    std::string sProgramName = argv[0];
+    bool        sIsSuccess   = true;
+
+    sf_client::ServerInfoV1      sSfServerV1;
+    sf_client::CommandArgsV1     sSfArgsV1;
+    sf_client::CommandResponseV1 sSfResponseV1;
+    sf_client::Session           sSession;
+
+    SfClientArgs sArgs;
+    sIsSuccess = ParseArgs(argc, argv, sArgs);
+
+    if(sArgs.mVerbose)
+    {
+        Verbose_PrintArgs(sArgs);
+    }
+
+    sIsSuccess = true;
```

Comment:
Should this be removed?  Argument parsing failures and not causing the help text to print